packages/hooks/src/hooks.spec.js

```diff
@@ -0,0 +1,120 @@
+import React, { useEffect } from 'react';
+import { act, renderHook } from 'react-hooks-testing-library';
+import { render, cleanup } from 'react-testing-library';
+import {
+  useDocumentEventListener,
+  useElementNode,
+  useViewportSize,
+} from './index';
+
+afterAll(cleanup);
+
+describe('packages/hooks', () => {
+  describe('useDocumentEventListener', () => {
+    test('event callback should fire when enabled is true', () => {
+      const eventCallback = jest.fn();
+
+      renderHook(() => useDocumentEventListener('click', eventCallback));
+
+      act(() => {
+        document.body.dispatchEvent(new MouseEvent('click', { bubbles: true }));
+      });
+
+      expect(eventCallback).toHaveBeenCalledTimes(1);
+    });
+
+    test('event callback should not fire when enabled is false', () => {
+      const eventCallback = jest.fn();
+
+      renderHook(() =>
+        useDocumentEventListener({
+          type: 'click',
+          eventCallback,
+          enabled: false,
+        }),
+      );
+
+      act(() => {
+        document.body.dispatchEvent(new MouseEvent('click', { bubbles: true }));
+      });
+
+      expect(eventCallback).toHaveBeenCalledTimes(0);
+    });
+  });
+
+  describe.skip('useMutationObserver', () => {}); //eslint-disable-line jest/no-disabled-tests
+
+  describe('useElementNode', () => {
+    let count = 1;
+
+    function TestUseElementNode() {
+      const [refEl, setRefEl] = useElementNode();
+      useEffect(() => {
+        count += 1;
+      }, [refEl]);
+      return <div ref={setRefEl} />;
+    }
+
+    render(<TestUseElementNode />);
+
+    test('it gets called twice', () => {
+      expect(count).toEqual(2);
+    });
+
+    const { result } = renderHook(() => useElementNode());
+
+    test('returns an array where the first element refers to the ref node', () => {
+      expect(result.current[0] === null).toBe(true);
+    });
+
+    test('returns an array where the second element is a callback to update the ref node', () => {
+      expect(typeof result.current[1]).toBe('function');
+    });
+  });
+
+  describe('useViewportSize', () => {
+    const { result, rerender } = renderHook(() => useViewportSize());
+
+    test('responds to updates in window width', () => {
+      act(() => {
+        window.width = 200;
+        window.dispatchEvent(new Event('resize'));
+        rerender();
+      });
+      setTimeout(() => {
+        expect(result.current.width).toBe(200);
+      }, 100);
+
+      act(() => {
+        window.width = 1024;
+        window.dispatchEvent(new Event('resize'));
+        rerender();
+      });
+
+      setTimeout(() => {
+        expect(result.current.width).toBe(1024);
+      }, 200);
+    });
+
+    test('responds to updates in window height', () => {
```

Comment:
This test isn't testing anything that the last one isn't. By testing both width and height, this second test really is only testing that the window resize event is working.

That said, if you can break the hook in a way that fails one of these tests and not the other, let me know!

Reply:
I understand what you're saying, but felt like stronger coverage to write both. I can remove the second test block if you prefer though.

Reply:
It's additional maintenance burden – since you're directly dispatching the resize event in both, that's what triggers the hook. I don't see any added benefit in having both in here.